app/src/main/java/com/example/ecodigify/notifications/NotificationScheduler.kt

```diff
@@ -0,0 +1,89 @@
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.os.Build
+import android.util.Log
+import androidx.core.app.NotificationCompat
+import com.example.ecodigify.dataclass.Ingredient
+import java.util.TimeZone
+
+class NotificationScheduler(private val context: Context) {
+    init {
+        createNotificationChannel()
+    }
+
+    // Schedule a notification for a specific LocalDate (on that date at midnight)
+    fun scheduleNotificationFor(ingredient: Ingredient) {
+        val alarmManager = context.getSystemService(Context.ALARM_SERVICE) as AlarmManager
+
+        val notificationTimeMillis = ingredient.expirationDate
+            .atStartOfDay()
+            .atZone(TimeZone.getDefault().toZoneId())
+            .toInstant()
+            .toEpochMilli()
+
+        // Create a PendingIntent to trigger the notification
+        val intent = Intent(context, NotificationReceiver::class.java)
+        // Put indredient name in intent
+        intent.putExtra("ingredient_name", ingredient.name)
+
+        val pendingIntent = PendingIntent.getBroadcast(
+            context,
+            ingredient.id.toInt(),
+            intent,
+            PendingIntent.FLAG_UPDATE_CURRENT
+        )
+
+        // Set the alarm to trigger at the specified date
+        alarmManager.setExact(AlarmManager.RTC_WAKEUP, notificationTimeMillis, pendingIntent)
+
+        Log.d(
+            "NotificationScheduler",
+            "Notification scheduled for ${ingredient.name} at ${ingredient.expirationDate} midnight"
+        )
+    }
+
+    // Create the Notification Channel (Required for Android 8.0 and above)
+    @SuppressLint("ObsoleteSdkInt")
+    private fun createNotificationChannel() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            val channel = NotificationChannel(
+                notificationChannelId,
+                "Scheduled Notifications",
+                NotificationManager.IMPORTANCE_DEFAULT
+            )
+            val notificationManager =
+                context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+            notificationManager.createNotificationChannel(channel)
+        }
+    }
+
+    fun scheduleNotificationForEach(ingredients: List<Ingredient>) {
+        ingredients.forEach(::scheduleNotificationFor)
+    }
+
+    // Function to create and send a notification
+    fun sendNotification(ingredientName: String) {
+        val notificationManager =
+            context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+
+        val notification = NotificationCompat.Builder(context, notificationChannelId)
+            .setSmallIcon(android.R.drawable.ic_notification_overlay)
+            .setContentTitle("Expiration Notice")
+            .setContentText("$ingredientName is about to going bad!")
```

Comment:
```suggestion
            .setContentText("$ingredientName is about to go bad!")
```

Reply:
sarebbe da spostare anche in string.xml i messaggi (giusto per rimanere fedeli allo standard)

Reply:
Penso che questo si applichi solo alla UI e a stringhe che vengono riutilizzate. I messaggi verranno usati solo da una funzione quindi direi che possiamo evitare